### PR TITLE
Streamline project docs and contributor guidance

### PR DESCRIPTION
**Pull request by frostney**
## Summary
- Reorganized `AGENTS.md` to make core contributor rules, GC notes, language restrictions, and design patterns easier to scan.
- Tightened and shortened the top-level docs in `README.md` while pushing detailed references into the dedicated pages under `docs/`.
- Updated several documentation pages to improve cross-links and keep the main entry points aligned with the current project layout.

## Testing
- Not run (documentation-only changes).
- Checked that the affected docs files were updated consistently across `AGENTS.md`, `README.md`, and `docs/`.

### CODE REVIEW

File: docs/built-ins.md [hunk 1]
```diff
@@ -2,6 +2,13 @@
 
 *For contributors adding or modifying built-in objects, and for script authors looking up available APIs.*
 
+## Executive Summary
+
+- **Flag-gated registration** — Each built-in is controlled by a `TGocciaGlobalBuiltins` flag (e.g., `ggConsole`, `ggMath`, `ggJSON`), enabling sandboxed execution
+- **22 built-in flags** — Console, Math, Object, Array, Number, JSON, JSON5, JSONL, TOML, YAML, Symbol, Set, Map, Promise, Performance, Temporal, JSX, ArrayBuffer, FFI, Proxy, Reflect, plus test/benchmark frameworks
```
[the remark this answers is not in the document]
Comment (frostney) [reply docs/built-ins.md]: Fixed — updated to 23. Verified against the actual `TGocciaGlobalBuiltin` enum in `Goccia.Engine.pas` (23 entries).

File: docs/bytecode-vm.md [hunk 2]
```diff
@@ -2,6 +2,13 @@
 
 *For contributors working on the bytecode backend.*
 
+## Executive Summary
+
+- **Two execution modes** — tree-walk interpreter (default) and bytecode VM (`--mode=bytecode`), sharing the same frontend, runtime objects, and GC
+- **Goccia-owned VM** — executes directly on `TGocciaValue` with tagged `TGocciaRegister` values; not a generic VM layer
+- **Opcode space** — core instructions (0-127) for hot paths, semantic instructions (128-255) for colder operations like imports/exports
```
[the remark this answers is not in the document]
Comment (frostney) [reply docs/bytecode-vm.md]: Fixed — updated to show the three-tier split: core (0-127), non-core generic ops (128-166), semantic/helper (167-255). Verified `OP_IMPORT = 167` in `Goccia.Bytecode.pas`.

File: docs/language-restrictions.md [hunk 3]
```diff
@@ -2,6 +2,13 @@
 
 *For script authors learning what GocciaScript supports and excludes, and for contributors understanding the design boundaries.*
 
+## Executive Summary
+
+- **Modern subset** — `let`/`const`, arrow functions, classes with private fields, `for...of`, async/await, ES modules (named only)
+- **Excluded by design** — `var`, `function` keyword, `==`/`!=`, `eval`, `arguments`, traditional loops, `with`, default imports/exports
+- **Graceful handling** — Excluded syntax parses successfully but executes as a no-op with a warning and suggestion
```
[the remark this answers is not in the document]
Comment (frostney) [reply docs/language-restrictions.md]: Fixed — narrowed to explicitly list the parser-recognized exclusions (`var`, `function`, `==`, loops, `with`). Other exclusions like `eval` and `arguments` are runtime errors, not parser no-ops.

File: docs/testing.md [hunk 4]
```diff
@@ -2,6 +2,13 @@
 
 *For contributors writing, running, or debugging tests.*
 
+## Executive Summary
+
+- **JavaScript tests are primary** — End-to-end `.js` tests in `tests/` are the source of truth; Pascal unit tests are secondary
+- **Built-in test framework** — `describe`/`test`/`expect` with async support, mock functions, lifecycle hooks, and Vitest-compatible matchers
+- **One method per file** — Each test file focuses on a single method; edge cases are co-located with happy-path tests
+- **Run with**: `./build.pas testrunner && ./build/TestRunner tests`
```
[the remark this answers is not in the document]
Comment (frostney) [reply docs/testing.md]: Fixed — added `--asi` to the command. CI runs `--asi` globally since ASI tests under `tests/language/asi/` require it.

File: docs/language-restrictions.md [hunk 5]
```diff
@@ -2,6 +2,13 @@
 
 *For script authors learning what GocciaScript supports and excludes, and for contributors understanding the design boundaries.*
 
+## Executive Summary
+
+- **Modern subset** — `let`/`const`, arrow functions, classes with private fields, `for...of`, async/await, ES modules (named only)
+- **Excluded by design** — `var`, `function` keyword, `==`/`!=`, `eval`, `arguments`, traditional loops, `with`, default imports/exports
+- **Graceful handling** — Parser-recognized excluded syntax (`var`, `function`, `==`, loops, `with`) parses successfully but executes as a no-op with a warning and suggestion
+- **Opt-in features** — ASI (`--asi`), JSX (`ggJSX`), types as comments, decorators, enums
```
[the remark this answers is not in the document]
Comment (frostney) [reply docs/language-restrictions.md]: Fixed — split into two bullets: opt-in toggles (ASI, JSX) vs always-available extensions (types as comments, decorators, enums). Verified none of the latter have `gg*` flags in `TGocciaGlobalBuiltin`.
